site/app/controllers/grading/ImagesController.php

```diff
@@ -181,4 +177,21 @@ public function ajaxUploadImagesFiles() {
         }
         return $this->core->getOutput()->renderResultMessage($message, true);
     }
+
+    /**
+     * @Route("/courses/{_semester}/{_course}/flag_user_image", methods={"POST"})
+     * @AccessControl(role="INSTRUCTOR")
+     */
+    public function flagUserImage() {
+        $result = $this->core->getQueries()->updateUserDisplayImageState($_POST['user_id'], 'flagged');
+
+        if ($result) {
+            $this->core->addSuccessMessage($_POST['user_id'] . '\'s image was successfully flagged.');
+            $this->core->redirect($this->core->buildCourseUrl(['student_photos']));
+        }
+        else {
+            $this->core->addErrorMessage('Some error occurred flagging ' . $_POST['user_id'] . '\'s image.');
+            $this->core->redirect($this->core->buildCourseUrl(['student_photos']));
```

Comment:
You can get rid of each redirect call here and instead return a response object after the else statement so it looks like

```php
return MultiResponse::RedirectOnlyResponse(
    new RedirectResponse($this->core->buildUrl(['student_photos']))
);
```

Reply:
Changed